src/main/java/hudson/remoting/Request.java

```diff
@@ -234,11 +234,14 @@ public boolean isDone() {
             public RSP get() throws InterruptedException, ExecutionException {
                 synchronized(Request.this) {
                     try {
-                        while(response==null) {
+                        while(response==null && !channel.isInClosed()) {
                             if (isCancelled()) {
                                 throw new CancellationException();
                             }
-                            Request.this.wait(); // wait until the response arrives
+                            Request.this.wait(30*1000); // wait until the response arrives
```

Comment:
@oleg-nenashev I would go the opposite of you and say "how can the limit be too small - its 30 seconds and is in a loop"?
I would hate to block to 5 seconds if the channel is being closed down - for me it is way too big :-)
calling `isCancelled()` and `channel.IsInClosed()` should be light enough that we can call them every second!

Anyway - 30 seconds is better than ∞